lib/sqlite_manager.rb

```diff
@@ -1,29 +1,25 @@
-require 'sqlite3'
-
 require_relative './row_transformer'
 
-# Client for creating and updating local sqlite3 database
+# Manages fetching and parsing records
 class SQLITEClient
-    def initialize
-        @db = SQLite3::Database.new "/tmp/#{ENV['SQLITE_FILE']}"
-    end
-
     def fetch_records(holding_id)
-        rows = exec_query "SELECT * FROM boxes WHERE record_num = #{holding_id}"
+        rows = []
+        offset = 0
+        limit = 100_000
+        loop do
+            $logger.info "Querying sierra for record batch #{offset}:#{limit}"
+            box_rows = $pg_client.exec_query(ENV['DB_QUERY'], holding_id, offset: offset, limit: limit)
```

Comment:
I think this `loop` might be unnecessary, we know (well, we think we know) that there can be only up to 120 cards per holding record, so we should be able to make this query once with confidence.